tests/test_early_stopping.py

```diff
@@ -235,7 +240,7 @@ class TestEarlyStoppingRealWorld(unittest.TestCase):
     #: The maximum number of epochs to train. Should be large enough to allow for early stopping.
     max_num_epochs: int = 1000
     #: The epoch at which the stop should happen. Depends on the choice of random seed.
-    stop_epoch: int = 21
+    stop_epoch: int = 30
```

Comment:
Changing the evaluation times is a somewhat breaking change, in the sense that when training exactly the same model with fixed random seed and early stopping than we did before this PR it will stop at a different epoch (which is why the `stop_epoch` changed). However, I would argue that evaluating at epoch `frequency`, `2* frequency` is more natural and what you would expect when setting an evaluation frequency, rather than `1`, `frequency + 1`, `2 * frequency + 1`, etc.

Reply:
this breaking change is fine with me. I think the new way makes  more sense